etcdmain/etcd.go

```diff
@@ -89,6 +89,11 @@ func startEtcdOrProxyV2() {
 	defaultHostOverride := defaultHost == "" || defaultHostErr == nil
 	if (defaultHostOverride || cfg.Name != embed.DefaultName) && cfg.InitialCluster == defaultInitialCluster {
```

Comment:
this path needs a lot of special support from Config and seems self-contained, can everything starting from `IsDefaultHost()` to the end of this `if` be `func (cfg *Config) UpdateDefaultClusterFromName()` instead?